aws/resource_aws_s3_bucket.go

```diff
@@ -734,6 +735,12 @@ func resourceAwsS3BucketUpdate(d *schema.ResourceData, meta interface{}) error {
 		// Retry due to S3 eventual consistency
 		_, err := retryOnAwsCode(s3.ErrCodeNoSuchBucket, func() (interface{}, error) {
 			terr := keyvaluetags.S3BucketUpdateTags(s3conn, d.Id(), o, n)
+			// S3BucketUpdateTags may wrap the underlying awserr.Error. Unwrap
+			// it here so that retryOnAwsCode can evaluate it properly.
+			var awsErr awserr.Error
```

Comment:
It might be better to go with a more general fix to the `retryOnAwsCode` helper. The problem was also fixed for me when I instead changed the lines [here](https://github.com/terraform-providers/terraform-provider-aws/blob/v2.59.0/aws/awserr.go#L43-L44) to:

```golang
var awsErr awserr.Error
if errors.As(err, &awsErr)&& awsErr.Code() == code {
```

I could modify this to use the more general fix but wasn't sure if this might have unintended effects for other resources that use this helper.